src/hjson_value.cpp

```diff
@@ -347,6 +347,27 @@ Value& Value::at(const char *name) {
   return at(std::string(name));
 }
 
+#if __cplusplus >= 201703L
+void Value::insert(const std::string&key, Value&& other)
+{
+  switch (prv->type)
+  {
+  case Type::Undefined:
+    prv->~ValueImpl();
+    // Recreate the private object using the same memory block.
+    new(&(*prv)) ValueImpl(Type::Map);
+    // Fall through
+  case Type::Map:
+    try {
+      prv->m->m.insert_or_assign(key, std::move(other));
+    } catch(const std::out_of_range&) {}    
+    return;
+  default:
+    throw type_mismatch("Must be of type Map for that operation.");
```

Comment:
The implementation seems to support type Undefined too, not just type Map.

Reply:
true, this implementation follows a similar scheme as to that of operator [] , where if the Value is of type undefined, we assume the user wanted map and thus we convert it to a map type and then treat it as a map insertion.